### PR TITLE
BLD: update extras to match new ci-helpers

### PR DESCRIPTION
**Pull request by tangkong**
## Description
Updates build requirements to match updated CI-Helpers from https://github.com/pcdshub/pcds-ci-helpers/pull/127

## Motivation and Context
Improve builds and testing in CI.  Also a small spelling test that I have been failing 😢 

## How Has This Been Tested?
Looking for green checks in CI

## Where Has This Been Documented?
This PR

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran ``docs/pre-release-notes.sh`` and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,18 +1,14 @@
 pytest
-doctr
+pytest-cov
 ipython
```
Comment (klauer) [inline dev-requirements.txt:3]: `ipython` is used only in the docs, isn't it?

File: docs/source/upcoming_release_notes/316-bld_pyproj_extras.rst
Comment (klauer) [inline docs/source/upcoming_release_notes/316-bld_pyproj_extras.rst:1]: (Unrelated to this file) Can you remove the documentation deploy key `.enc` that was used for Travis CI? Looks like we missed it in the last migration.

File: dev-requirements.txt [hunk 2]
```diff
@@ -1,18 +1,14 @@
...
 pymongo
 mongomock >=3.22.0
-sphinx
-sphinx_rtd_theme
 # Removed temporarily, pip installations fail lacking gssapi libs
 # psdm_qs_cli
-sphinx-click
 # PyPI's 2020 resolver can be problematic with these
 # dependencies-of-dependencies.  Constrain them here:
 six >=1.10.0
```
Comment (klauer) [inline dev-requirements.txt:12]: I wonder if we still need this

File: dev-requirements.txt (hunk 1 above)
Comment (tangkong) [reply dev-requirements.txt:3]: We also use it in the [cli tests](https://github.com/pcdshub/happi/blob/a985ac79407593cf022ddbc1b5eb4aff305dfa14/happi/tests/test_cli.py#L12).

File: dev-requirements.txt (hunk 2 above)
Comment (tangkong) [reply dev-requirements.txt:12]: Now's as good a time as any to check 👀

Comment (tangkong) [reply dev-requirements.txt]: Seems like it's no longer necessary 👍   At least on CI

**Review by klauer — APPROVED** [review]

**Pull request merged**